pkg/destroy/azure/azure.go

```diff
@@ -467,16 +467,16 @@ func deleteResourceGroup(ctx context.Context, client resources.GroupsClient, log
 
 	delFuture, err := client.Delete(ctx, name)
 	if err != nil {
-		if wasNotFound(delFuture.Response()) {
+		if delFuture.FutureAPI != nil && wasNotFound(delFuture.Response()) {
 			logger.Debug("already deleted")
 			return nil
 		}
-		return err
+		return errors.Wrapf(err, "failed to delete %s", name)
 	}
 
 	err = delFuture.WaitForCompletionRef(ctx, client.Client)
 	if err != nil {
-		if wasNotFound(delFuture.Response()) {
+		if delFuture.FutureAPI != nil && wasNotFound(delFuture.Response()) {
```

Comment:
The added check for a non-nil `FutureAPI` is not needed here. The contract with the `client.Delete` function is that `delFuture` will have a non-nil `FutureAPI` unless an error was returned. Regardless, if the code got to this point, then `FutureAPI` cannot be nil, otherwise the call to `delFuture.WaitForCompletionRef` would have raised a panic.